### PR TITLE
fix(appveyor): fix up failing unit tests on Windows

### LINKED ISSUE #2456: [META] Works on Windows, yet?
Helm has some bugs when it comes to working on and with Windows. Filing as a high level ticket to track windows support, unless there is another way.

* [x] #2362 Helm init failed on windows
* [x] #2396 Failed to install vendor dependencies on Windows
* [x] #2420 Support Tiller installs in clusters with Windows nodes
* [ ] #1606 Add AppVeyor CI support for Windows testing
* [x] #2071 Error: open .helm\repository\cache\local-index.yaml
* [x] #2252 Error using repo add

There are more issues still to be determined.

### LINKED ISSUE #3690: helm repo index with subfolders creates urls with backslashes
Hello

We try to setup a chart registry and planned to structure it with subfolders. So here an example that represents what we tried:

```
$ tree .
.
|-- cockroachdb
|   |-- cockroachdb-0.5.4.tgz
|   `-- cockroachdb-0.6.6.tgz
`-- postgresql
    |-- postgresql-0.8.13.tgz
    |-- postgresql-0.9.0.tgz
    |-- postgresql-0.9.1.tgz
    `-- postgresql-0.9.2.tgz

2 directories, 6 files
```

Then create the index:
```
$ helm repo index --url https://gitserver/scm/k8s/charts/raw .
$ cat index.yaml
```

This results in the urls having a "%5C" in them (I believe a backslash):
```
...
    urls:
    - https://gitserver/scm/k8s/charts/raw/postgresql%5C/postgresql-0.9.0.tgz
    version: 0.9.0
  - created: 2018-03-15T13:47:43.2792366+01:00
    description: Object-relational database management system (ORDBMS) with an emphasis
      on extensibility and on standards-compliance.
    digest: be307fd5fab1b6e2af67b23d6a17dbdad66ee2c7ce2d166b04d1f36f3e1555dd
    engine: gotpl
    home: https://www.postgresql.org/
    icon: https://www.postgresql.org/media/img/about/press/elephant.png
    keywords:
    - postgresql
    - postgres
    - database
    - sql
    maintainers:
    - name: swordbeta
    - name: databus23
    name: postgresql
    sources:
    - https://github.com/kubernetes/charts
    - https://github.com/docker-library/postgres
    urls:
    - https://gitserver/scm/k8s/charts/raw/postgresql%5C/postgresql-0.8.13.tgz
    version: 0.8.13
generated: 2018-03-15T13:47:43.2767378+01:00
```

using helm version 2.8.2 on windows.
this might not evevn be supported. I found some closed PR that should allow recursive index creation and slashes in chart names. So not sure if this is intentional. Since all commands work without complaints I see it as a bug.

We will use a flat structure for now - so nothing urgent. If all our services become helm charts we easily will have around 500 charts with 0-40 versions of each chart per week.

### PR DESCRIPTION
**Pull request by bacongobbler**
Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

closes #2456
closes #3690

### CODE REVIEW

**Comment by bacongobbler** [comment]
woop! We have legitimate Windows test failures. 🎉

**Comment by bacongobbler** [comment]
closes #2456

**Comment by mattfarina** [comment]
@bacongobbler This is failing because it needs a `make docs` according to circleci.

**Comment by bacongobbler** [comment]
@mattfarina yes, the debug code is there while I figure out what's causing the last test case to fail. I'm in Belgium with no access to a Windows box 😢 🐼

**Comment by bacongobbler** [comment]
Also RE the `make docs` failure, that was a merge fail in master that I fixed in #4924 yesterday. I'll go ahead and rebase this now :)

**Comment by bacongobbler** [comment]
Okay, Windows tests have gone green and it looks like I've fixed #3690 at the same time via de6aece. Ready for review!

File: pkg/chartutil/requirements_test.go [hunk 1]
```diff
@@ -426,7 +428,12 @@ func TestDependentChartWithSubChartsHelmignore(t *testing.T) {
 }
 
 func TestDependentChartsWithSubChartsSymlink(t *testing.T) {
-	c, err := Load("testdata/joonix")
+	joonix := filepath.Join("testdata", "joonix")
+	if err := os.Symlink(filepath.Join("..", "..", "frobnitz"), filepath.Join(joonix, "charts", "frobnitz")); err != nil {
```
Comment (bacongobbler) [inline pkg/chartutil/requirements_test.go]: note for the reviewer: there appears to be a bug where symlinking a directory on Linux, committing that change and then testing [if it's a symbolic link](https://github.com/helm/helm/blob/3a8a797eab0e1d02456c7944bf41631546ee2e47/pkg/sympath/walk.go#L112-L115) on a Windows box fails. The fix for this is to symlink the directory on a Windows box for Windows testing. That seems to work to get the tests passing again, and I don't think there's a good way to fix this.

File: pkg/chartutil/requirements_test.go [hunk 2]
```diff
@@ -25,22 +27,22 @@ import (
 )
 
 func TestLoadRequirements(t *testing.T) {
-	c, err := Load("testdata/frobnitz")
+	c, err := Load(filepath.Join("testdata", "frobnitz"))
```
Comment (adamreese) [inline pkg/chartutil/requirements_test.go]: Another option would be adding a `filepath.FromSlash(...)' inside `Load()` to convert to os separators

Comment (bacongobbler) [reply pkg/chartutil/requirements_test.go]: Yeah that'd work! Both approaches are good, though your suggestion should work fine and reduces LOC changed. 👍

Comment (bacongobbler) [reply pkg/chartutil/requirements_test.go]: fixed.

**Comment by radu-matei** [comment]
I think it has to do with how `git` is configured on the Windows box (specifically the behaviour of Git Bash when using PowerShell), but:

- when using PowerShell:

```
 $ go.exe test k8s.io\helm\pkg\getter -run ^TestPluginGetter$
--- FAIL: TestPluginGetter (0.03s)
    plugingetter_test.go:83: exec: "echo": executable file not found in %PATH%
FAIL
FAIL    k8s.io/helm/pkg/getter  0.093s
```

- when using Git Bash:

```
$ go.exe test k8s.io/helm/pkg/getter -run ^TestPluginGetter$
ok      k8s.io/helm/pkg/getter  0.160s
```

This boils down to `echo` being a command under Windows, and not an executable, as for UNIX systems. So while the AppVeyor tests might be passing (since most likely they run under Git Bash), under regular PowerShell `TestPluginGetter` fails.

**Comment by bacongobbler** [comment]
okay. I've skipped that particular test because there is no equivalent to `echo` on Windows, as it's a builtin command for cmd.exe.

**Review by adamreese — APPROVED** [review]

**Review by radu-matei — APPROVED** [review]
LGTM

**Pull request merged**